src/mailmap.rs

```diff
@@ -0,0 +1,207 @@
+      Self::Owned(mailmap) => mailmap,
+    }
+  }
+}
+
+impl DerefMut for MailmapInner {
+  fn deref_mut(&mut self) -> &mut Self::Target {
+    match self {
+      Self::Repo(repo) => repo.deref_mut(),
+      Self::Owned(mailmap) => mailmap,
+    }
+  }
+}
+
+/// A wrapper around git2::Mailmap providing Node.js bindings
+#[napi]
+pub struct Mailmap {
+  pub(crate) inner: MailmapInner,
+}
+
+#[napi]
+impl Mailmap {
+  /// Create a mailmap from the contents of a string.
+  ///
+  /// The format of the string should follow the rules of the mailmap file:
+  /// ```
+  /// # Comment line (ignored)
+  /// Seokju Me <seokju.me@toss.im> Seokju Na <seokju.me@gmail.com>
+  /// ```
+  ///
+  /// @param {string} content - Content of the mailmap file
+  /// @returns A new mailmap object or null if operation failed
+  #[napi]
+  pub fn from_buffer(content: String) -> crate::Result<Option<Mailmap>> {
+    match git2::Mailmap::from_buffer(&content) {
+      Ok(mailmap) => Ok(Some(Mailmap {
+        inner: MailmapInner::Owned(mailmap),
+      })),
+      Err(e) => Err(e.into()),
+    }
+  }
+
+  /// Add a new Mailmap entry.
+  ///
+  /// Maps an author/committer (specified by `replace_name` and `replace_email`)
+  /// to the specified real name and email. The `replace_email` is required but
+  /// the other parameters can be null.
+  ///
+  /// If both `replace_name` and `replace_email` are provided, then the entry will
+  /// apply to those who match both. If only `replace_name` is provided,
+  /// it will apply to anyone with that name, regardless of email. If only
+  /// `replace_email` is provided, it will apply to anyone with that email,
+  /// regardless of name.
+  ///
+  /// @param {string} [real_name] - The real name to use, or null
+  /// @param {string} [real_email] - The real email to use, or null
+  /// @param {string} [replace_name] - The name to replace, or null
+  /// @param {string} replace_email - The email to replace
+  /// @returns true if the operation succeeded, false otherwise
+  #[napi]
+  pub fn add_entry(
+    &mut self,
+    real_name: Option<String>,
+    real_email: Option<String>,
+    replace_name: Option<String>,
+    replace_email: String,
+  ) -> bool {
+    match &mut self.inner {
+      MailmapInner::Repo(repo) => repo
+        .add_entry(
+          real_name.as_deref(),
+          real_email.as_deref(),
+          replace_name.as_deref(),
+          &replace_email,
+        )
+        .is_ok(),
+      MailmapInner::Owned(mailmap) => mailmap
+        .add_entry(
+          real_name.as_deref(),
+          real_email.as_deref(),
+          replace_name.as_deref(),
+          &replace_email,
+        )
+        .is_ok(),
+    }
+  }
+
+  /// Resolve a signature to its canonical form using a mailmap.
+  ///
+  /// Returns a new signature with the canonical name and email.
+  ///
+  /// @param {SignaturePayload} signature - Signature to resolve
+  /// @returns The resolved signature with canonical name and email
```

Comment:
We would also appreciate it if you could provide JSDoc documentation for the Mailmap implementation. 👍 
Required jsdocs are: `@category`, `@signature`